sync3/conn.go

```diff
@@ -78,11 +78,15 @@ func (c *Conn) OnUpdate(ctx context.Context, update caches.Update) {
 }
 
 func (c *Conn) tryRequest(ctx context.Context, req *Request) (res *Response, err error) {
+	// TODO: include useful information from the request in the sentry hub/context
 	defer func() {
 		panicErr := recover()
 		if panicErr != nil {
 			err = fmt.Errorf("panic: %s", panicErr)
 			logger.Error().Msg(string(debug.Stack()))
+			internal.GetSentryHubFromContextOrDefault(ctx).RecoverWithContext(ctx, panicErr)
+		} else if err != nil {
+			internal.GetSentryHubFromContextOrDefault(ctx).CaptureException(err)
```

Comment:
Why are these calling different things? What is the difference between them? We just want to send up the error, nothing more, so `CaptureException` seems the right thing to do?

Reply:
Commented in https://github.com/matrix-org/sliding-sync/pull/57/commits/646232dcb0e6f6daa82dc8dd7770cbf121bac001.

Reply:
But TL;DR: I think sentry might mark `panicErr` as having come from a panic with the former function.